src/System.Windows.Forms.Design/tests/UnitTests/System/Resources/Tools/StronglyTypedResourceBuilderTests.cs

```diff
@@ -0,0 +1,402 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.CodeDom.Compiler;
+using System.Collections;
+using System.ComponentModel;
+using System.Drawing;
+using Microsoft.CSharp;
+using Newtonsoft.Json.Linq;
+using Xunit;
+using static Interop.Gdi32;
+
+namespace System.Resources.Tools.Tests;
+
+public partial class StronglyTypedResourceBuilderTests
+{
+    // https://docs.microsoft.com/dotnet/core/extensions/work-with-resx-files-programmatically
+
+    private static readonly CodeDomProvider s_cSharpProvider = new CSharpCodeProvider();
+
+    private static string CreateResx(string data = null) =>
+        $"""
+        <root>
+            <resheader name="resmimetype">
+                <value>text/microsoft-resx</value>
+            </resheader>
+            <resheader name="version">
+                <value>2.0</value>
+            </resheader>
+            <resheader name="reader">
+                <value>System.Resources.ResXResourceReader</value>
+            </resheader>
+            <resheader name="writer">
+                <value>System.Resources.ResXResourceWriter</value>
+            </resheader>
+            {data ?? string.Empty}
+        </root>
+        """;
+
+    [Fact]
+    public void Create_NullBaseName_ThrowsArgumentNull()
```

Comment:
Please follow the test naming convention: https://github.com/dotnet/winforms/blob/main/docs/testing.md#naming
```suggestion
    public void StronglyTypedResourceBuilder_Create_NullBaseName_ThrowsArgumentNull()
```

Reply:
This seems redundant and makes the TestExplorer a pain to use, but I'll change.